File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        n = 200
        for s in strs:
            n = min(n, len(s))
            
        for i in range(n):
            fix = strs[0][i]
            flag = True
            for s in strs:
                if fix != s[i]:
                    flag = False
                    break
            if flag:
                prefix += fix
            else:
                break
                
        return prefix

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < strs[0].length(); i++) {
            char c = strs[0].charAt(i);
            boolean done = false;
            for (String str : strs) {
                if (i >= str.length() || c != str.charAt(i)) {
                    done = true;
                    break;
                }
            }
            if (done) {
                break;
            }
            sb.append(c);
        }

        return sb.toString();
    }
}